python/pyarrow/tests/test_parquet.py

```diff
@@ -1843,18 +1843,41 @@ def test_filters_read_table(tempdir):
     assert table.num_rows == 3
 
 
-@pytest.yield_fixture
-def s3_example():
-    access_key = os.environ['PYARROW_TEST_S3_ACCESS_KEY']
-    secret_key = os.environ['PYARROW_TEST_S3_SECRET_KEY']
-    bucket_name = os.environ['PYARROW_TEST_S3_BUCKET']
+@pytest.fixture
+def s3_bucket(request, minio_server):
+    boto3 = pytest.importorskip('boto3')
+    botocore = pytest.importorskip('botocore')
+
+    address, access_key, secret_key = minio_server
+    s3 = boto3.resource(
+        's3',
+        endpoint_url='http://{}'.format(address),
+        aws_access_key_id=access_key,
+        aws_secret_access_key=secret_key,
+        config=botocore.client.Config(signature_version='s3v4'),
+        region_name='us-east-1'
+    )
+    bucket = s3.Bucket('test-s3fs')
+    bucket.create()
+    return 'test-s3fs'
+
 
-    import s3fs
-    fs = s3fs.S3FileSystem(key=access_key, secret=secret_key)
+@pytest.fixture
+def s3_example(minio_server, s3_bucket):
+    s3fs = pytest.importorskip('s3fs')
+
+    address, access_key, secret_key = minio_server
+    fs = s3fs.S3FileSystem(
+        key=access_key,
+        secret=secret_key,
+        client_kwargs={
+            'endpoint_url': 'http://{}'.format(address)
+        }
+    )
 
     test_dir = guid()
+    bucket_uri = 's3://{0}/{1}'.format(s3_bucket, test_dir)
 
-    bucket_uri = 's3://{0}/{1}'.format(bucket_name, test_dir)
     fs.mkdir(bucket_uri)
```

Comment:
Are the S3-Parquet tests actually working? If yes, then nice :)

Reply:
Yep, it works.